src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,40 @@
-import React from 'react';
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface Props {
+  selectedTodo: Todo | null,
+  onClose: () => void,
+}
+
+export const TodoModal: React.FC<Props> = (props) => {
+  const { selectedTodo, onClose } = props;
+  const [isModalLoading, setIsModalloading] = useState(false);
+
+  const [
+    currentUser,
+    setCurrentUser,
+  ] = useState<User | null>(null);
+
+  useEffect(() => {
+    setIsModalloading(true);
+
+    getUser(selectedTodo?.userId ?? 0)
```

Comment:
And this one check will be redundant.
Also, you shouldn't call API with invalid userId at all